### PR TITLE
DOCS-2632 Monitoring HTTP Traffic Redirects to HTTPS Guide

### PR DESCRIPTION
**Pull request by HugoPct**
### What does this PR do?

Adds a guide to introduce a best practice for HTTP tests.

### Motivation

DOCS-2632

### Preview

https://docs-staging.datadoghq.com/guide/http-redirection/synthetics/guide/monitor-https-redirection

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by HugoPct** [comment]
the 2 screenshots are
1. https://a.cl.ly/RBuENg5d
2. https://a.cl.ly/geupk5gZ

File: content/en/synthetics/guide/monitor-https-redirections.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    - link: '/synthetics/http_tests'
+      tag: 'Documentation'
+      text: 'Create a HTTP Test'
+
+---
+
+Monitoring that your HTTP traffic is redirected into HTTPS is key to ensure the encryption of your users connections with your API endpoints and your application.
+
+## Monitor your HTTPS redirections
+Depending on your setup, you can find information about the redirection either in the response header with the presence of the `location` header or in the response body from your HTTP request (presence of `"https:"===window.location.protocol`).
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirections.md:14]:
```suggestion

Depending on your setup, you can identify the redirect to HTTPS in the generated **Response Preview** tab under Headers as `location` or in the **Body** as `"https:"===window.location.protocol`.
```

File: content/en/synthetics/guide/monitor-https-redirections.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+To start monitoring the redirection of your HTTP traffic into HTTPS:
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirections.md:16]:
```suggestion
To monitor the redirection of your HTTP traffic into HTTPS:
```

File: content/en/synthetics/guide/monitor-https-redirections.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+2. Click on the **Test Url** button and look for redirection information in the response preview.
+3. Make an assertion on the presence of the redirection element.
+
+<!-- Screenshot 1 et 2-->
+
+You will then be alerted if your HTTP traffic is not correctly redirected into HTTPS.
+
+## Further Reading
+
+{{< partial name="whats-next/whats-next.html" >}}
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirections.md:27]:
```suggestion
{{< partial name="whats-next/whats-next.html" >}}

[1]: /getting_started/synthetics/api_test/#define-request

```

File: content/en/synthetics/guide/monitor-https-redirections.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+Monitoring that your HTTP traffic is redirected into HTTPS is key to ensure the encryption of your users connections with your API endpoints and your application.
+
+## Monitor your HTTPS redirections
+Depending on your setup, you can find information about the redirection either in the response header with the presence of the `location` header or in the response body from your HTTP request (presence of `"https:"===window.location.protocol`).
+
+To start monitoring the redirection of your HTTP traffic into HTTPS:
+1. Create an HTTP.
+2. Click on the **Test Url** button and look for redirection information in the response preview.
+3. Make an assertion on the presence of the redirection element.
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirections.md:19]:
```suggestion

1. Create an HTTP test and [define the request][1].
2. Click **Test URL **. The response preview generates a **Request Preview** and **Response Preview**.
3. Click **+ New Assertion** or a `location` header to define an assertion on the redirection element.
```

File: content/en/synthetics/guide/monitor-https-redirections.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+<!-- Screenshot 1 et 2-->
+
+You will then be alerted if your HTTP traffic is not correctly redirected into HTTPS.
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirections.md:23]:
```suggestion
Datadog alerts you if your HTTP traffic does not correctly redirected into HTTPS.
```

**Review by alai97 — CHANGES_REQUESTED** [review]

**Review by alai97 — APPROVED** [review]

**Comment by HugoPct** [comment]
Change a bit the PR. 
So it is more clear that they have to choose either asserting on the headers or the body

File: content/en/synthetics/guide/monitor-https-redirection.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+### Monitor your HTTPS redirection
+
+Depending on your setup, you can identify the redirect to HTTPS in the generated **Response Preview** tab under Headers as `location` or in the **Body** as `"https:"===window.location.protocol`.
+
+To monitor the redirection of your HTTP traffic into HTTPS:
+
+1. Create an HTTP test and [define the request][1].
+2. Click **Test URL**. The response preview generates a **Request Preview** and **Response Preview**.
+3. Assert on the redirection
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirection.md:23]:
```suggestion
3. Add an assertion about the redirection to HTTPS.
```

File: content/en/synthetics/guide/monitor-https-redirection.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    - Define an assertion on the `location` header.  
+    e.g.: under **Headers**, the `location` header for `http://datadoghq.com` is `https://datadoghq.com`.
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirection.md:25]:
```suggestion
    - Define an assertion on the `location` header.  For example, under **Headers**, the `location` header for `http://datadoghq.com` is `https://datadoghq.com`.
    
```

File: content/en/synthetics/guide/monitor-https-redirection.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    {{< img src="synthetics/guide/monitor-https-redirections/location-header-https.png" alt="Location header in the response preview" style="width:100%;" >}}
+    - or select `body` `contains` and insert `"https:"===window.location.protocol` in the text field.
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirection.md:27]:
```suggestion
    - Optionally, select `body` `contains` and paste `"https:"===window.location.protocol` in the text field. 
```

File: content/en/synthetics/guide/monitor-https-redirection.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+3. Assert on the redirection
+    - Define an assertion on the `location` header.  
+    e.g.: under **Headers**, the `location` header for `http://datadoghq.com` is `https://datadoghq.com`.
+    {{< img src="synthetics/guide/monitor-https-redirections/location-header-https.png" alt="Location header in the response preview" style="width:100%;" >}}
+    - or select `body` `contains` and insert `"https:"===window.location.protocol` in the text field. 
+    {{< img src="synthetics/guide/monitor-https-redirections/https-assertion.png" alt="Define your assertion" style="width:100%;" >}}
+
+
+
+Complete the rest of the test creation workflow and save your HTTP test. After defining the notification, Datadog can alert you when your HTTP traffic does not correctly redirect into HTTPS.
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirection.md:32]:
```suggestion
Complete the rest of the test creation workflow and save your HTTP test. 

After defining the notification, Datadog can alert you when your HTTP traffic does not correctly redirect into HTTPS.
```

File: content/en/synthetics/guide/monitor-https-redirection.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+1. Create an HTTP test and [define the request][1].
+2. Click **Test URL**. The response preview generates a **Request Preview** and **Response Preview**.
+3. Assert on the redirection
+    - Define an assertion on the `location` header.  
+    e.g.: under **Headers**, the `location` header for `http://datadoghq.com` is `https://datadoghq.com`.
+    {{< img src="synthetics/guide/monitor-https-redirections/location-header-https.png" alt="Location header in the response preview" style="width:100%;" >}}
+    - or select `body` `contains` and insert `"https:"===window.location.protocol` in the text field. 
+    {{< img src="synthetics/guide/monitor-https-redirections/https-assertion.png" alt="Define your assertion" style="width:100%;" >}}
+
+
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirection.md:30]:
```suggestion
```

**Review by alai97 — CHANGES_REQUESTED** [review]

**Review by alai97 — APPROVED** [review]

File: content/en/synthetics/guide/monitor-https-redirection.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+To monitor the redirection of your HTTP traffic into HTTPS:
+
+1. Create an HTTP test and [define the request][1].
+2. Click **Test URL**. The response preview generates a **Request Preview** and **Response Preview**.
+3. Add an assertion about the redirection to HTTPS.
+    - Define an assertion on the `location` header. For example, under **Headers**, the `location` header for `http://datadoghq.com` is `https://datadoghq.com`.
+    
+    {{< img src="synthetics/guide/monitor-https-redirections/location-header-https.png" alt="Location header in the response preview" style="width:100%;" >}}
+    - Optionally, select `body` `contains` and paste `"https:"===window.location.protocol` in the text field.
```
Comment (HugoPct) [inline content/en/synthetics/guide/monitor-https-redirection.md:27]: it is really either assert on header "location" OR assert on the body.
So I feel "optionally" is not the right word

Comment (alai97) [reply content/en/synthetics/guide/monitor-https-redirection.md:27]:
```suggestion
    - Alternatively, define an assertion on the response body by clicking **+ New Assertion**. Select `body` `contains` and paste `"https:"===window.location.protocol` in the text field. 
```

File: content/en/synthetics/guide/monitor-https-redirection.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+### Monitor your HTTPS redirection
+
+Depending on your setup, you can identify the redirect to HTTPS in the generated **Response Preview** tab under Headers as `location` or in the **Body** as `"https:"===window.location.protocol`.
+
+To monitor the redirection of your HTTP traffic into HTTPS:
+
+1. Create an HTTP test and [define the request][1].
+2. Click **Test URL**. The response preview generates a **Request Preview** and **Response Preview**.
+3. Add an assertion about the redirection to HTTPS.
+    - Define an assertion on the `location` header. For example, under **Headers**, the `location` header for `http://datadoghq.com` is `https://datadoghq.com`.
```
Comment (alai97) [inline content/en/synthetics/guide/monitor-https-redirection.md:24]:
```suggestion
    - Define an assertion on the `location` header by clicking the `location` header in the response preview. For example, under **Headers**, the `location` header for `http://datadoghq.com` is `https://datadoghq.com`.
```

**Review by alai97 — APPROVED** [review]

**Pull request merged**